libc/src/stdio/printf_core/writer.h

```diff
@@ -22,60 +22,86 @@ namespace LIBC_NAMESPACE_DECL {
 namespace printf_core {
 
 struct WriteBuffer {
+  enum class WriteMode {
+    FILL_BUFF_AND_DROP_OVERFLOW,
+    FLUSH_TO_STREAM,
+    RESIZE_AND_FILL_BUFF,
+    BUTT,
```

Comment:
I like using an enum over function pointers, since a branch to direct call is more easily optimized out than an indirect function.
```suggestion
    BUTT,
```
Assuming this is a debug metavalue?

Reply:
What does buttmode do?